### DISCUSSION TITLE
Docs: Supporting Next.js App deployment to China

### DISCUSSION THREAD

**Discussion by Moto-Venturer** [vercel/next.js#39693] [Ideas]
### What is the improvement or update you wish to see?

I created a PR ([vercel/next.js: Pull Request 39228](https://github.com/vercel/next.js/pull/39228)) a few days ago about supporting Next.js app deployment for China, and @leerob mentioned pushing it back on the issue stage, so I'm creating this issue here for other devs to review.

### Is there any context that might help us understand?

A quick background brief: Mainland China is unlike any other country. Vercel or Netlify-like deployment solutions don't exist. [Vercel](https://www.21cloudbox.com/solutions/how-to-speed-up-vercel-in-china.html#:~:text=Does%20Vercel%20work%20in%20China%3F) and [Netlify](https://www.21cloudbox.com/solutions/alternatives-to-netlify-in-china.html) don't work out-of-the-box in China due to their Firewall and other legal issues.

21YunBox created [guides](https://www.21cloudbox.com/solutions/how-to-speed-up-vercel-in-china.html) to make deployment of Next.js apps easy like Vercel for China. If you think this is good conjunction for the Nextjs app in China, I think adding some links on the [next.js docs](https://nextjs.org/docs/deployment) would definitely help developers get less headache for deployment in China.

Do leave comments here, so I know what you guys think.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/deployment

**Comment by SukkaW** [comment]
Personally, I'd vote no for adding 21YunBox to the Next.js documentation.

As shown for the 21YunBox [pricing page](https://www.21cloudbox.com/pricing.html), even the cheapest `starter` plan is for 38600 CNY/year (which is 6433 USD/year). Unlike Vercel, Netlify, and Firebase have completely free hobby plans or DigitalOcean and AWS have reasonable prices for developers, 21YunBox is definitely an enterprise-only platform, which also means it is not for everyone.

See also:

- Hugo has concerns before adding 21YunBox to their documentation: https://github.com/gohugoio/hugoDocs/pull/1319

cc @leerob

**Reply by begaaam** [reply]
je suis débutant avec Nextjs, comment je peux installer une ancienne version de NEXT, par exemple la version : NEXT11.1.4
npx create-next-app@11.1.4, ça ne marche pas . merci

**Reply by PrimozRome** [reply]
I was in touch with 21YunBox and they guys offered to fix our Chinese hostin with Next.js and DatoCMS content for 140k$ per year... So much about their service...

**Comment by Moto-Venturer** [comment]
China's web ecosystem is not like any other country.  The average non-Chinese developer is [legally](https://www.21cloudbox.com/solutions/how-to-get-an-ICP-license-for-china.html) restricted from hosting their projects in China. In comparison, foreign companies can typically afford the [cost of hosting a website in China legally](https://www.21cloudbox.com/cost-to-host-a-website-in-china.html).

However, 21YunBox is far from being an enterprise-only platform.  There are many developers (some foreign and many Chinese) using their [non-commercial plans](https://www.21cloudbox.com/non-commercial-use-plans.html).  Many Chinese developers are even using 21YunBox to get their Next.js projects working in China.

**Comment by SukkaW** [comment]
As requested, I have checked [non-commercial plans](https://www.21cloudbox.com/non-commercial-use-plans.html). The "outstanding" "Agency" plan requires 3800 CNY/year (that's 633 USD/year). So no, 21YunBox is still not an option for everyone, and it doesn't support developers by any means.

Also, [all your case studies](https://www.21cloudbox.com/case-studies.html) are fake. None of the companies the website mentioned are actually using 21YunBox to provide a better service in Mainland China. IMHO, **I am afraid you are attempting to use the Next.js website for advertising your platform** that almost nobody uses. You are not here trying to share more options with developers.

> what I mean for nobody:
> Search term `deploy Next.js on Netlify` has 415000 results, the term `deploy Next.js DigitalOcean` has 214000 results, while the term `deploy Next.js on "21YunBox"` only has 1900 results and they are all your advertisements:
> <img width="735" alt="image" src="https://user-images.githubusercontent.com/40715044/184803401-d195d6a9-149e-4419-9d8f-a4d233a1be96.png">
> <img width="710" alt="image" src="https://user-images.githubusercontent.com/40715044/184804161-3627ce5b-604f-4470-b091-e6ee7a2f3561.png">
> <img width="768" alt="image" src="https://user-images.githubusercontent.com/40715044/184803494-f6540820-7db5-4234-9f46-fc722caa1c75.png">

> And the search results get even worse on the Chinese search engine Baidu:
> <img width="748" alt="image" src="https://user-images.githubusercontent.com/40715044/184805079-89a66910-6a61-4389-b13b-568571439e08.png">
> <img width="784" alt="image" src="https://user-images.githubusercontent.com/40715044/184805039-00bee53a-ee0c-4d56-bd83-6c7c98b80849.png"> 

Another thought (not Next.js documentation related): Your company's website `www.21yunbox.com` is hosted in Hong Kong, not in Mainland China. And your domain `21yunbox.com` do not have any valid ICP License. Besides, 21YunBox does not have the MIIT Class B IDC License (which is essential for any company to provide hosting service in Mainland China). How are you trying to convince us that your service is trustworthy?

**Reply by Moto-Venturer** [reply]
@SukkaW, First of all, thanks for your passion for keeping Next.js free from spammers and scams.

Sure you are right, 21YunBox may not be as globally recognized as Netlify.  Particularly as Jamstack technologies like Next.js are just starting to be more adopted in China 👍.  But they are focused on providing oversea developers trying to deploy sites in China with a technology they are familiar with.

As for the hits by Hugo or Next.js, search in Baidu or Google. Every company got to start somewhere. As far as I know, 21YB has started out small and fully privately owned, so they don't serve a "free plan," unlike many other VC-funded companies. so that's why there are limited search hits on the search engines.

Free and cheap solutions are great, but that's typically only for individuals. Companies and enterprises want reliable and easy solutions. They need someone who can talk their language and help them with their journeys in China, so it's a matter of purchase preference. There is no right or wrong in any sense.

I believe Next.js is a global and open platform. I don't see a reason why @leerob and the Next.js admins would not be welcoming company and enterprise service providers to bridge the gap from oversea tech to China.

**Comment by Moto-Venturer** [comment]
To provide more context for those who haven't faced the China deployment issue yet. 

> _Mainland China is unlike any other country. Vercel or Netlify-like deployment solutions don't exist (aside from 21YunBox). [Vercel](https://www.21cloudbox.com/solutions/how-to-speed-up-vercel-in-china.html#:~:text=Does%20Vercel%20work%20in%20China%3F) and [Netlify](https://www.21cloudbox.com/solutions/alternatives-to-netlify-in-china.html) don't work out-of-the-box in China due to their Firewall and other legal issues._

  I ran a quick speed test on https://nextjs.org (deployed with Vercel) and compared it with a fast Jamstack site deployed with 21YunBox.

(Red, orange, and yellow = Slow, green = Fast)
<img width="1232" alt="Screen Shot 2022-08-23 at 9 24 58 AM" src="https://user-images.githubusercontent.com/78116874/186198978-0ef5f0d5-19f8-4925-888a-f4ec495fa5fc.png">

The speed map on the left shows that Nextjs.org is slow and unstable across much of China.  

What do you think? Would it be beneficial for the Next.js community to have access to a deployment solution to bridge the gap between oversea tech to China?

**Comment by Moto-Venturer** [comment]
Worth mentioning that 21YunBox loves and supports the open-source community.  They offer a free plan to provide stable and fast-access for open-source projects in China.

<img width="1746" alt="Screen Shot 2022-10-21 at 2 48 22 PM 1" src="https://user-images.githubusercontent.com/78116874/197286209-2602d890-4a5f-486b-ac1a-7dc7b4c52181.png">

**Reply by Zhangyinshuai** [reply]
Can you give me an invitation slot？, I registered and did not reply。
![微信截图_20220915135454](https://user-images.githubusercontent.com/22345204/190325506-b620d856-ba02-48ff-a9a3-a31e0ddadbe1.png)

**Comment by PrimozRome** [comment]
I was in touch with 21YunBox and they guys offered to fix our Chinese hosting with Next.js and DatoCMS content for 140k$ per year... If you want to go that way be my guest.
